app/controllers/sessions_controller.rb

```diff
@@ -48,16 +42,15 @@ def account_manager_url
     Plek.find("account-manager")
   end
 
-  def fetch_http_referrer
-    http_referrer = request.headers["HTTP_REFERER"]
-
-    return nil unless http_referrer&.start_with?(Plek.new.website_root)
-
-    http_referrer.delete_prefix Plek.new.website_root
-  end
+  def http_referer_path
+    @http_referer_path ||=
+      begin
```

Comment:
Interesting there's no rescue here - and you are using safe navigators etc.
So I can't see a case where an error could bubble up?

In this scenario what does the `begin` do for us?

Reply:
It lets us use a multi-line block of code, `@http_referer_path` ends up with the value of:

```ruby
http_referer = request.headers["HTTP_REFERER"]
if http_referer&.start_with?(Plek.new.website_root)
  http_referer.delete_prefix Plek.new.website_root
end
```

So, `/foo` if it's of the form `https://www.gov.uk/foo` and `nil` otherwise